### ISSUE TITLE
centos box and private network lost IP

### ISSUE THREAD

**Issue by gcuisinier** [mitchellh/vagrant#6235]
With this vagrant file :

```
Vagrant.configure(2) do |config|

  config.vm.box = "centos7"
  config.vm.hostname = "docker-registry"
  config.vm.box_url = "https://github.com/holms/vagrant-centos7-box/releases/download/7.1.1503.001/CentOS-7.1.1503-x86_64-netboot.box"

  #config.vm.network "public_network"
  config.vm.network "private_network", ip: "192.168.100.20"
  #config.vm.synced_folder ".", "/vagrant", type: "nfs"

  # VirtualBox
  config.vm.provider :virtualbox do |vb|
    vb.name = "poc_docker"
    vb.memory = 3072
    vb.cpus = 2
    vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
    vb.customize ["modifyvm", :id, "--natdnsproxy1", "on"]
  end


end
```

The ping respond first at "192.168.100.20", but failed after some time.
When I connect on the box, it seems that the ip is lost.

```
sudo /sbin/ifup eth1
```
or a reboot fix the problem.

But what's the root cause ? Any ideas ?
Where am I wrong in my Vagrantfile ?

**Comment by sethvargo** [comment]
Hi @gcuisinier 

Thank you for opening an Issue, and I am sorry you are experiencing problems with Vagrant. I just ran your exact Vagrantfile on Vagrant 1.7.4, Virtualbox 5.0.8 r103449 for over 2 hours and the box is still pingable.

Are you able to share any additional information, such as the [debug logs](https://docs.vagrantup.com/v2/other/debugging.html) from the Vagrant run?

**Comment by gcuisinier** [comment]
Hi,

I will try to reproduce, but I've upgraded my virtualbox. So I am not sure it is always present.
The problem was that Vagrant fixes the IP, but the network service was not reloaded, so for CentOS, the interface was alwaws manage by DHCP ...
And after dhcp timeout, the network interface was putting down.

This provision script fixed my problem : 

```
sudo nmcli connection reload
sudo systemctl restart network.service
```

**Comment by coutotyler** [comment]
I'm seeing a similar issue. After some time the private network interface doesn't have an IP address. I've also noticed that the network service is down. I'm trying to run github.com/cloudbase/salt-openstack on this which requires the NetworkManager to be off.

**Comment by sethvargo** [comment]
This has been fixed. I forget the issue, but we now are very careful with nmcli.

**Comment by sethvargo** [comment]
Will be released in Vagrant 1.8!
